reco_utils/dataset/wikidata.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import requests
+
+def find_wikidataID(name):
+    """Find the entity ID in wikidata from a title string.
+    Args:
+        name (str): A string with a wikidata page title.
+    Returns:
+        entityID: wikidata entityID corresponding to the title string. 
+                  'entityNotFound' will be returned if no page is found
+    """
+    url = "https://en.wikipedia.org/w/api.php?action=query&prop=pageprops&format=json&titles="
```

Comment:
Hardcoded string may not be desirable. Make it an input variable or constant.

Reply:
hi, sorry for the delay in the answer. since the request to the new query (I had to do some changes) is a concatenation of strings and a variable:

`requests.get("https://en.wikipedia.org/w/api.php?action=query&list=search&srsearch="+name+"&format=json&prop=pageprops&ppprop=wikibase_item")` 

how do you suggest I can do this? should make the two substrings constants and concatenate them in the function?

Reply:
here's an approach
```
# defined at beginning of module
import urllib
API_URL = "https://en.wikipedia.org/w/api.php"


def find_wikidataID(name):
    url_opts = "&".join([
        "action=query",
        "list=search",
        "srsearch={}".format(urllib.parse.quote(name)),
        "format=json",
        "prop=pageprops",
        "ppprop=wikibase_item",
    ])
    requests.get("{url}?{opts}".format(url=API_URL, opts=url_opts))
```

Reply:
I implemented this here: a5338c0 Thanks a lot for the tip!